monai/data/synthetic.py

```diff
@@ -148,7 +148,7 @@ def create_test_image_3d(
         else:
             image[circle] = rs.random() * 0.5 + 0.5
 
-    labels = np.ceil(image).astype(np.int32)
+    labels = np.ceil(image).astype(np.int32, copy=False)
```

Comment:
there is a `dtype` option in `np.ceil` not sure if we reuse it. it's the same for np.multiply, np.ascontiguousarray, etc.

Reply:
Hi @wyli ,

Thanks for your review.
I know that arg `dtype` in `ceil`, `multiply`, etc. But the doc-string of numpy said it's the data type for `computation`.
https://numpy.org/doc/stable/reference/ufuncs.html#ufuncs-kwargs
_"Overrides the DType of the output arrays the same way as the signature. This should ensure a matching precision of the calculation. The exact calculation DTypes chosen may depend on the ufunc and the inputs may be cast to this DType to perform the calculation."_
Here we convert the data type after computation, so I didn't use that arg.

Thanks.

Reply:
Ok, how about np.ascontiguousarray? and I think the casting of np.random.normal should be at least floating numbers.

Reply:
Sure, I updated the `np.ascontiguousarray()` with dtype in the latest commit.
And the casting of np.random.normal in this line is `float32`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/spatial/array.py#L1305
Or do you mean other logic?

Thanks.

Reply:
this one is not correct https://github.com/Project-MONAI/MONAI/blob/33b6d61fdd8a15247e63bf7db16ac32f98f83566/monai/transforms/intensity/array.py#L171-L172 when the images are int arrays, the noise should be floats, but this is a bug introduced earlier

Reply:
Hi @wyli ,

I updated the PR with `dtype` option, similar idea to other intensity transforms.
What do you think?

Thanks.